game/game/gamescript.cpp

```diff
@@ -18,47 +18,32 @@
 #include "gothic.h"
 
 using namespace Tempest;
-using namespace Daedalus::GameState;
 
-struct GameScript::ScopeVar final {
-  ScopeVar(Daedalus::DaedalusVM& vm,Daedalus::PARSymbol& sym,Npc& n)
-    :ScopeVar(vm,sym,n.handle(),Daedalus::IC_Npc){
-    }
-
-  ScopeVar(Daedalus::DaedalusVM& vm,Daedalus::PARSymbol& sym,Npc* n)
-    :ScopeVar(vm,sym,n ? n->handle() : nullptr, Daedalus::IC_Npc){
-    }
-
-  ScopeVar(Daedalus::DaedalusVM& vm, const char* name, Daedalus::GEngineClasses::C_Npc* h)
-    :ScopeVar(vm,vm.getDATFile().getSymbolByName(name),h,Daedalus::IC_Npc){
-    }
-
-  ScopeVar(Daedalus::DaedalusVM& vm,Daedalus::PARSymbol& sym, Daedalus::GEngineClasses::Instance* h, Daedalus::EInstanceClass instanceClass)
-    :vm(vm),sym(sym) {
-    prev = sym.instance;
-    sym.instance.set(h,instanceClass);
+template <typename T>
+struct ScopeVar final {
+  ScopeVar(phoenix::symbol* sym, std::shared_ptr<T> h) : prev(sym->get_instance()), sym(sym) {
```

Comment:
> phoenix::symbol* sym

Null symbol can't be used here anyway: `phoenix::symbol& sym` ?

`const std::shared_ptr<T>&` ? since copy-constructor of shared_ptr is non-zero cost.